adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralRTBBannerAdLoader.m

```diff
@@ -0,0 +1,141 @@
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationBannerAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+        
+    UIViewController *rootViewController = adConfiguration.topViewController;
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    if (!adUnitId.length || !placementId.length) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID or Placement ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+
+    NSError *error = nil;
+    CGSize bannerSize = [self bannerSizeFromAdConfiguration:adConfiguration error:&error];
+    if (error) {
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+
+    _bannerAdView = [[MTGBannerAdView alloc]initBannerAdViewWithAdSize:bannerSize placementId:placementId unitId:adUnitId rootViewController:rootViewController];
+    _bannerAdView.delegate = self;
+    _bannerAdView.autoRefreshTime = 0;
+    [_bannerAdView loadBannerAdWithBidToken:adConfiguration.bidResponse];
+}
+
+/// Returns the closest valid banner size by comparing the provided ad size against the valid sizes.
+/// Returns CGSizeZero and sets |error| if the ad configuration contains an invalid ad size.
+- (CGSize)bannerSizeFromAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
+                                  error:(NSError **)errorPtr {
+    GADAdSize adSize320x50 = GADAdSizeFromCGSize(CGSizeMake(320, 50));
+    GADAdSize adSize320x100 = GADAdSizeFromCGSize(CGSizeMake(320, 100));
+    GADAdSize adSize300x250 = GADAdSizeFromCGSize(CGSizeMake(300, 250));
+    GADAdSize adSize728x90 = GADAdSizeFromCGSize(CGSizeMake(728, 90));
+    NSArray<NSValue *> *possibleSizes = @[ @(adSize320x50), @(adSize320x100),  @(adSize300x250), @(adSize728x90)];
+    
+    GADAdSize requestedSize = adConfiguration.adSize;
+    GADAdSize closestAdSize = GADClosestValidSizeForAdSizes(requestedSize, possibleSizes);
+    
+    if (GADAdSizeEqualToSize(closestAdSize, GADAdSizeInvalid)) {
+        NSString *errorMessage = [NSString
+                                  stringWithFormat:
+                                      @"The requested banner size: %@ is not supported by Mintegral SDK.",
+                                  NSStringFromGADAdSize(requestedSize)];
+        *errorPtr = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegtalErrorBannerSizeInValid, errorMessage);
+        return CGSizeZero;
+    }
+    
+    return CGSizeMake(closestAdSize.size.width, closestAdSize.size.height);
+}
```

Comment:
Can you move this to [GADMAdapterMintegralUtils.m](https://github.com/googleads/googleads-mobile-ios-mediation/blob/main/adapters/Mintegral/MintegralAdapter/GADMAdapterMintegralUtils.m) since it will be used in both bidding and waterfall?

Reply:
ok,I have modified.